src/postprocess/chi0.jl

```diff
@@ -88,6 +86,32 @@ function compute_χ0(ham; droptol=0, temperature=ham.basis.model.temperature)
     χ0
 end
 
+# Solves Q (H-εn) Q δψn = -Q rhs
+# where Q is the projector on the orthogonal of ψk
+function sternheimer_solver(basis, kpoint, Hk, ψk, ψnk, εnk, rhs, cgtol)
+    # we err on the side of caution here by applying Q *a lot*
+    # there are optimizations to be made here
+    Q(ϕ) = ϕ - ψk * (ψk' * ϕ)
+    function QHQ(ϕ)
+        Qϕ = Q(ϕ)
+        Q(Hk * Qϕ - εnk * Qϕ)
+    end
+    precon = PreconditionerTPA(basis, kpoint)
+    precondprep!(precon, ψnk)
+    function f_ldiv!(x, y)
+        ldiv!(x, precon, Q(y))
+        x .= Q(x)
```

Comment:
I'm kind of tempted to write this in one line: `f_ldix!(x, y) = x .= Q(ldiv!(x, precon, Q(y)))`